risc0/build/src/lib.rs

```diff
@@ -908,19 +923,43 @@ mod tests {
     #[test]
     fn escapes_strings_when_encoding_when_requested() {
         let guest_meta = GuestMetadata {
-            rustc_flags: Some(RUSTC_FLAGS.iter().map(ToString::to_string).collect()),
+            rustc_flags: Some(vec!["--cfg".to_string(), "\"foo=bar\"".to_string()]),
             ..Default::default()
         };
         let encoded = encode_rust_flags(&guest_meta, true);
-        let expected = [
-            "--cfg",
-            "foo=\\\"bar\\\"",
-            "--cfg",
-            "foo=\\\'bar\\\'",
-            "-C",
-            "link-args=--fatal-warnings",
-        ]
-        .join("\x1f");
+        let expected = ["--cfg", "\\\"foo=bar\\\""].join("\x1f");
         assert!(encoded.contains(&expected));
     }
+
+    /// Test that MinGuestListEntry creates the same .bin path as GuestListEntry
+    /// This validates the fix for the issue where embed_method_metadata_with_options
+    /// was returning ELF paths instead of .bin paths
+    #[test]
+    fn min_guest_list_entry_creates_bin_path() {
+        // Create a temporary ELF file for testing
+        let temp_dir = std::env::temp_dir();
+        let test_elf_path = temp_dir.join("test_guest");
+        std::fs::write(&test_elf_path, b"dummy elf content").unwrap();
+
+        let guest_info = GuestInfo {
+            options: GuestOptions::default(),
+            metadata: GuestMetadata {
+                kernel: false, // This ensures we create the .bin file
+                ..Default::default()
+            },
+        };
+
+        // For this test, we'll test the logic without actually calling the build method
+        // since it requires proper ELF files and kernel files
+        let elf_path_str = test_elf_path.to_str().unwrap();
+        let expected_bin_path = format!("{}.bin", elf_path_str);
```

Comment:
please put the format arg in the string, this will soon be a clippy enforced error